### PR TITLE
fix transaction counter for "to" only addresses

### PR DESCRIPTION
**Pull request by r-gochain**

### CODE REVIEW

File: server/backend/backend.go [hunk 1]
```diff
@@ -272,7 +273,14 @@ func (b *Backend) GetAddressByHash(ctx context.Context, hash string) (*models.Ad
 	if err != nil {
 		return nil, err
 	}
-	address.NumberOfTransactions = int(nonce)
+	transactionCounter := int(nonce)
```
Comment (treeder) [inline server/backend/backend.go:276]: Is nonce ever the right value?  Incoming transactions won't be counted if we use the nonce right?

Comment (r-gochain) [reply server/backend/backend.go]: Updated, now using for everything

**Comment by treeder** [comment]
Would it make more sense to increment a counter rather than doing a count query every request?

**Comment by r-gochain** [comment]
@treeder If I'm not mistaken that was one of the first version (counter), the problem is that counter was constantly getting off (and it's hard to backfill the data) so we decided to use intermediate collection (TransactionsByAddress)

**Pull request merged**